python/btcspv/validate_spv.py

```diff
@@ -0,0 +1,170 @@
+    headerHash = rutils.hash256(header.raw)
+    if headerHash != header.hash_le:
+        # TODO: Throw error instead of returning false
+        # throw new Error('Hash LE is not the correct hash of the header')
+        return False
+
+    # Check that HashLE is the reverse of Hash
+    reversedHash = header.hash[::-1]
+    if reversedHash != header.hash_le:
+        # throw new Error('HashLE is not the LE version of Hash');
+        return False
+
+    # Check that the MerkleRootLE is the correct MerkleRoot for the header
+    extractedMerkleRootLE = extract_merkle_root_le(header.raw)
+    if extractedMerkleRootLE != header.merkle_root_le:
+        # throw new Error(
+        # 'MerkleRootLE is not the correct merkle root of the header')
+        return False
+
+    # Check that MerkleRootLE is the reverse of MerkleRoot
+    reversedMerkleRoot = header.merkle_root[::-1]
+    if reversedMerkleRoot != header.merkle_root_le:
+        # throw new Error('MerkleRootLE is not the LE version of MerkleRoot');
+        return False
+
+    # Check that PrevHash is the correct PrevHash for the header
+    extractedPrevHash = extract_prev_block_be(header.raw)
+    if extractedPrevHash != header.prevhash:
+        # throw new Error(
+        # 'Prev hash is not the correct previous hash of the header');
+        return False
+
+    return True
+
+
+def validate_spvproof(proof: object) -> bool:
+    '''
+    Verifies an SPV proof object
+    Args:
+        proof (object): The SPV Proof as an object
+        proof.vin (bytes): The vin
+        proof.vout (bytes): The vout
+        proof.locktime (bytes): The locktime
+        proof.tx_id (bytes): The tx ID
+        proof.tx_id_le (bytes): The LE tx ID
+        proof.index (int): The index
+        proof.intermediate_nodes (bytes): The intermediate nodes
+        proof.confirming_header.raw (bytes): The bitcoin header
+        proof.confirming_header.hash (bytes): The hash of the header
+        proof.confirming_header.hash_le (bytes): The LE hash
+        proof.confirming_header.height (int): The height
+        proof.confirming_header.prevhash (bytes): The hash of the
+            previous header
+        proof.confirming_header.merkle_root The merkle root of the header
+        proof.confirming_header.merkle_root_le The LE merkle root
+    Returns:
+        (bool): True if valid proof, else False
+    '''
+    validVin = validate_vin(proof.vin)
+    if not validVin:
+        # throw new Error('Vin is not valid')
+        return False
+
+    validVout = validate_vout(proof.vout)
+    if not validVout:
+        # throw new Error('Vout is not valid')
+        return False
+
+    # txID = calculateTxId(version, vin, vout, locktime)
+    txID = rutils.hash256(
+        proof.confirming_header['version'][2:] +
+        proof.confirming_header['vin'][2:] +
+        proof.confirming_header['vout'][2:] +
+        proof.confirming_header['locktime'][2:]
+    )
+    if txID != proof.tx_id_le:
```

Comment:
use `snake_case` for all variable names. `tx_id`

Reply:
Fixed